src/core/monitoring.h

```diff
@@ -20,124 +20,140 @@
 #include "fortelist.h"
 #include "conn.h"
 #include "stringdict.h"
-#include "datatypes/forte_array.h"
-#include "datatypes/forte_struct.h"
 
 class CFunctionBlock;
 class CResource;
 
-namespace forte {
-  namespace core {
+namespace forte::core {
+  namespace internal {
 
-    /*!\brief class that handles all monitoring tasks
-     *
-     */
-    class CMonitoringHandler {
+    class CWatchEntry {
       public:
-        explicit CMonitoringHandler(CResource &paResource);
+        CStringDictionary::TStringId getPortId() const {
+          return mPortId;
+        }
 
-        EMGMResponse executeMonitoringCommand(SManagementCMD &paCommand);
+        CWatchEntry(const CWatchEntry &) = delete;
+        CWatchEntry &operator=(const CWatchEntry &) = delete;
+
+      protected:
+        CWatchEntry(CStringDictionary::TStringId paPortId) : mPortId(paPortId) {
+        }
+
+        CWatchEntry(CWatchEntry &&) = default;
+        CWatchEntry &operator=(CWatchEntry &&) = default;
+
+      private:
+        CStringDictionary::TStringId mPortId;
+    };
+
+    class CDataWatchEntry : public CWatchEntry {
+      public:
+        CDataWatchEntry(CStringDictionary::TStringId paPortId, CIEC_ANY &paDataValue, TAbsDataPortNum paForceIndex) :
+            CWatchEntry(paPortId),
+            mDataBuffer(std::unique_ptr<CIEC_ANY>(paDataValue.clone(nullptr))),
```

Comment:
The intermediate temporary should not be necessary:
```suggestion
            mDataBuffer(paDataValue.clone(nullptr)),
```

Reply:
done